extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/JsonWebKeySet.java

```diff
@@ -27,6 +27,7 @@ public class JsonWebKeySet {
     private Map<String, Key> keysWithThumbprints = new HashMap<>();
     private Map<String, Key> keysWithS256Thumbprints = new HashMap<>();
     private Map<String, List<Key>> keysWithoutKeyIdAndThumbprint = new HashMap<>();
+    private List<Key> keys = new ArrayList<>();
```

Comment:
It should be a map IMHO, key is the algorithm, if JWK does not even have an algorithm property then we just ignore such keys for the purpose of the try-all fallback. Otherwise we may end up running EC keys against RSA signatures etc etc which will have time and possibly security  consequences

Reply:
In the implementation of the PR only keys which have a supported type will be stored.

Using a map with the algorithm as the key wouldn't be practical, as we need the list of keys while we only have a string representation of the jwt, so we cannot easily determine the algorithm and therefore the list of keys to use.